### ISSUE TITLE
Exception logging in before_sleep_log looses information on exception printing

### ISSUE THREAD

**Issue by himikof** [jd/tenacity#169]
The `before_sleep_log` function is very convenient, as it allows you to understand that an action took longer than expected due to retries from the log files. But the function currently just prints the exception that was the cause of retry as if calling `str(e)`.
Unfortunately, this way of printing can be very non-helpful (the following prints "raised ."):
```
try:
    assert False == True
except Exception as e:
    print("raised %s." % e)
```

Can you consider using one of alternative ways of formatting the exception?

1. Using `traceback.format_exception_only` method.
2. Using the built-in logging keyword argument `exc_info`.
3. Or just using `repr(e)` to log exception type and arguments.

**Comment by jd** [comment]
`traceback.format_exception_only` sounds neat for logging.

**Comment by jpark712** [comment]
@himikof This is fixed with #227.

You can add a `exc_info=True` parameter to the before_sleep_log function
@jd This issue can be closed.

**Comment by himikof** [comment]
Thanks for the update, this issue has indeed been solved now!
